---
layout: base_slides
topic_name: "galaxy_initiation_handsOn"
tutorial_name: "galaxy_initiation_handsOn"
---

.left.top[<img src="../images/LogoIP-CNRS-C3BI-NBV4small-e1460524231316.png" width="200">]

.center[#Galaxy training @ Institut Pasteur
##**From command line to web interface**
####Olivia Doppelt-Azeroual, Fabien Mareuil]

---
name: plan

## Plan

* Quick introduction to Galaxy; a workflow managment software

* Part 1 - **The web interface**:
    * Home page, tools, history, tool interface.
    * Data Library pannel
    * Workflow building pannel

* Part 2 - **Data analysis**:
    * Data upload:
        * data upload using 'upload file'
	    * data upload using 'shared data'
    * Job launching

* Part 3 - **Workflow construction**:
    * NGS workflow

---

## PART 1
## Quick introduction to Galaxy

---
layout: false
## Galaxy : a workflow managment software
* Galaxy is a web interface providing scientists, tools usually launched through command lines 
* **Basics** :
    * A galaxy tool is a program linked to a xml file
    * The XML file gathers options, types, inputs, outputs, formats and the tool command line.
    * Galaxy interprets the xml to generate the web interface 
* **Advantages++** :
    * Deals with Big Data
    * Creates workflows (tool chains) 
    * Enables saving and sharing workflows and data
* Galaxy@Pasteur is available at **galaxy.pasteur.fr**:
    * It is available for internal users since March 2013 and external users since October 2016
    * It is linked to the Institut Pasteur storage bay, enabling big data upload
    * We provide support for tools developement and tools installation in Galaxy

---
## Galaxy presentation: Homepage

<img src="../images/galaxy_presentation.png">

---
## Tool interface in Galaxy

<img src="../images/tool_interface.png">

---
## Data libraries management panel

<img src="../images/data_libraries.png">

---
## Workflow construction panel

<img src="../images/workflow.png">

---
name: connexion
## Hands on

* Connexion to Galaxy:
    - https://galaxy.pasteur.fr (linux/mac/windows)
    - if you do not have an account, register (menu User/Register)
    - email address (you will receive an email to activate your account):
        --> use your pasteur short address
    - password: as you want
    - public name: as you want

* Exploration of the 3 panels: 
    - Tools
    - History (a new one is created at each connexion)
    - Central panel

* Exploration of the 3 tabs:
    - Analyze Data
    - Workflow
    - Shared Data

---


## Part 2
## Data analysis


---
layout: false
## Upload a data
<img src="../images/upload.png">

* Use the tool **Upload File**
    * Data in Galaxy have a type/format
    * Pay a good attention to the file format and the selected Genome (for genomic data)

* Remark
    * This tool is adapted for dataset below 2 Go

---
name: upload_small
## Hands on

* Go to Galaxy

    - Search the category "Get Data" in the tool list
    - Select "upload file"  
      --> import the two fastq files

    - The two fastq file should be in the current history

---
name: launch
## Launch a job
<img src="../images/launch_job.png">
---
## Hands on - Select a tool in the tool list
* Category *NGS: QC and manipulation*, tool: *fqduplicate*
    - No input possible because it needs fastqsanger format and not fastq

* Solution: Choose the *fqconvert* tool in the category *Convert Formats*
    - Run *fqconvert* on one of the fastq input files

* Go back to *fqduplicate*
    - Run with *single read* parameter
---
name: tools
## Hands on - Select a tool in the tool list
* *fqextract*: parameters
    - Type of input data : single read
    - Original *fastq* dataset from your history: *fastqsanger* output from *fqconvert*
    - *Fastq* format : *illumina1.8/Sanger*
    - Ignore pair information for the extraction (-p): True
    - Dataset from your history containing the list of read names to extract or exclude (-l): *fqduplicate* from *fastqsanger*
    - Extract or exclude reads in the list: Extract listed reads
    - run
* Category NGS: Mapping: Parallel Map with BWA for Illumina
    * Will you select a reference genome from your history or use a built-in index?: Use one from the history
    * You need the reference genome,  warning: It can be a big file
---
name:big_upload
## Upload a data file bigger than 2Go

Command line solution
* Copy your data in your Galaxy transfer directory (use the SFTP protocol)
    * To establish the connection with the sftp server from your Linux or Mac computer:  
    ```bash
    sftp -P 2222 yourlogin@galaxy.pasteur.fr
    ```
    * To establish the connection with the sftp server from central-bio, tars or bic:  
    ```bash
    ssh yourlogin@central-bio.pasteur.fr 
    or
    ssh yourlogin@bic.pasteur.fr 
    or 
    ssh yourlogin@tars.pasteur.fr  
    sftp -P 2222 yourlogin@galaxy.pasteur.fr
    ```
.center[**REMARK: Use the galaxy password for the sftp connection**]
---
## Upload a data file bigger than 2Go

**Command line solution**
* To upload a file on the sftp server:
```bash
sftp>put PathToYourFile
```
* To upload a directory on the sftp server:
    * on the sftp server create a directory with the same name that the directory which you want to upload:
    ```bash
    sftp>mkdir mydirectory
    sftp>put -r mydirectory
    ```

---
## Upload a data file bigger than 2Go

**Filezilla solution**
* You can use filezilla to transfert data from your computer in your Galaxy transfer directory 
    * Launch filezilla
    * open the site manage: ctrl+s 
    * click on new site
        * host: galaxy.pasteur.fr
        * port: 2222
        * Protocol: sftp - ssh File Transfer Protocol
        * Logon Type: Ask for password
        * User: yourlogin
    * drag'n drop your data from the left windows to the right windows

.center[**REMARK: don't forget to use your galaxy password**]
---
## Upload a data file bigger than 2Go

* Then use the galaxy web interface to transfer the data in your data library and then to one of your history 
    * *Shared Data* interface
<img src="../images/shared_data.png">
    * Click on your library then on *Add datasets*
        * It's possible to create folder to organize your data

---
## Upload a data file bigger than 2Go

* Select the option *upload directory of files*
    * You can select the genome species

<img src="../images/upload_data.png">

---


